### PR TITLE
RadioControl: Use CSS-in-JS

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Use CSS-in-JS for `RadioControl`.

Note: I discovered a bug documented here while working on this: https://github.com/WordPress/gutenberg/issues/26293. This is a pre-existing bug that I'm not sure exactly how to fix. I think it should be done in the context of a separate pull request once this one is merged and once there is an answer about the intention for the radio control in the bug issue.

## How has this been tested?
Verified there are no differences between this branch and production in Storybook. Also checked in Gutenberg itself.

Captura de Tela 2020-10-27 às 08 22 16

## Types of changes
Non-breaking CSS-in-JS refactor. All classnames are preserved allowing folks to continue targeting the classes that were already there.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/radio-control/styles/radio-control-styles.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		}
+	}
+
+	&:focus {
+		box-shadow: 0 0 0 calc( ${ config( 'borderWidth' ) } * 2 ) ${ color(
+	'white'
+) },
+			0 0 0 calc( ${ config( 'borderWidth' ) } * 2 + ${ config(
+	'borderWidthFocus'
+) } )
```
Comment (saramarcondes) [inline packages/components/src/radio-control/styles/radio-control-styles.js:67]: `prettier` sure did a number on these lines. Not sure if it's worth trying to do anything about it.

Comment (ItsJonQ) [reply packages/components/src/radio-control/styles/radio-control-styles.js:67]: REKT. I'm seeing that on my side as well. Prettier is super unhappy with this. I don't blame it... there are a lot of things going on 🤣 

Maybe we can create a mixin for it. It'll make it easier to read (kinda), but it should make the linters happier.

Something like this:

```js
const radioFocusBoxShadow = () => {
	const boxShadow = [
		`0 0 0 calc( ${ config( 'borderWidth' ) } * 2 ) ${ color( 'white' ) }`,
		`0 0 0 calc( ${ config( 'borderWidth' ) } * 2 + ${ config(
			'borderWidthFocus'
		) } ) var( --wp-admin-theme-color )`,
	].join( ',' );

	return css( { boxShadow } );
};
```

**Review by ItsJonQ — COMMENTED** [review]
@saramarcondes Haiii! I left a suggestion for verbose box-shadow vs Prettier 😂

File: packages/components/src/radio-control/styles/radio-control-styles.js (hunk 1 above)
Comment (ItsJonQ) [reply packages/components/src/radio-control/styles/radio-control-styles.js]: It looks like the linter is still somewhat confused by this. I guess I don't blame it.
Totally not your fault! This makes me think that we need a better strategy (maybe) to handle complex string interpolation for CSS shorthands especially if it involves functions like `calc()`.

**Comment by ItsJonQ** [comment]
@saramarcondes We're starting to venture into components that require references to the `var( --wp-admin-theme-color )` variable. We currently don't have an automated way to handle this yet, which means it won't work currently in IE11.

I left some more thoughts here:
https://github.com/WordPress/gutenberg/pull/26340#issuecomment-717952369

I don't think there's a good solution yet, other than to manually hardcode the fallbacks 🙈 

We could create a "mixin" that can output the property for use.. something like:

```
const StyledInput = styled.input`
  ${variable('background', 'wp-admin-theme-color', 'blue')}
`
```

That's kind of how you'd do it in Sass. I **really don't** like this solution. It's quite limited (e.g. handling multiple values) and feels... too abstract.  But it is an option if we don't want to hand-write fallbacks.

Would love your thoughts!

**Comment by saramarcondes** [comment]
@ItsJonQ Shall we compromise and update `ui.theme` in the colors values to be the following:

```js
css`
  color: blue;
  color: var( --... );
`;
```

**Comment by ItsJonQ** [comment]
@saramarcondes We can certainly give that a shot 🎉

**Comment by saramarcondes** [comment]
I'll update this in the FormToggle PR as it's older and I'm hoping to get it merged sooner 🙂

**Comment by saramarcondes** [comment]
Closing in favor of focusing on G2-components integration.

**Pull request closed without merging**